modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairAbstractFuture.java

```diff
@@ -128,17 +138,26 @@ protected GridNearReadRepairAbstractFuture(
         this.expiryPlc = expiryPlc;
         this.tx = tx;
 
+        assert strategy != null;
+
+        this.strategy = strategy;
+
         canRemap = topVer == null;
 
-        map(canRemap ? ctx.affinity().affinityTopologyVersion() : topVer);
+        this.topVer = canRemap ? ctx.affinity().affinityTopologyVersion() : topVer;
     }
 
     /**
-     * @param topVer Affinity topology version.
+     *
      */
-    protected synchronized void map(AffinityTopologyVersion topVer) {
-        this.topVer = topVer;
+    protected void init() {
+        map();
+    }
 
+    /**
+     *
+     */
+    protected synchronized void map() {
```

Comment:
`onResult` and `map` should be synced because can be run in an async way from different threads.
This going to be simplified/refactored at IGNITE-16071.